### ISSUE TITLE
New fields to financial features (Suggestion)

### ISSUE THREAD

**Issue by arv187** [nabeelio/phpvms#957]
I think the fields for the database should be modified to add future features. For some it will be necessary to calculate the time (in minutes) at the gate/parking from the acars. Anyway some of them may be shortcut with #776

I have made a very primitive draft but that would give solution to future improvements such as #758 #776 #589 

[draft in excel](https://docs.google.com/spreadsheets/d/1vyel5t3NprTBUrF_l72IrSujQINaji5XcynI_CEVsao/edit?usp=sharing)

I want to try to learn to help or create a module with this stuff, but I have these doubts that I asked by discord:

To create an addon are these instructions? [https://github.com/nabeelio/phpvms#development-environment ](https://github.com/nabeelio/phpvms#development-environment) and [https://docs.phpvms.net/developers/environment](https://docs.phpvms.net/developers/environment)
[14:06]
What is the difference between addon and module?

**Comment by airnubeiro** [comment]
Hi! I´ve got an idea on how we could implement the fact to add the "fees per time" option. I don´t have a lot of coding skills, so I´ll try to explain it with my own words. Hopefully understandable :)

As already mentioned, normally airports will apply the fees depending on your MTOW and, on some fees, depending on the time you use a service. Let´s take the parking fees for example. AENA in Spain gives you the fees of all airports. There are a lot of tables from where you can take the factor you´ll have to multiply your MTOW. We could then maybe do something like this:

When you finish your flight, this airplane will stay at another airport that isn´t your hub. From this moment let´s start a "CRONO TIMER" until this airplane (we can take the airplane´s ID or REG or whatever) is used for another flight. The time that the airplane has beed on ground would be mutliplyed by:

a.) A specified number that has been introduced in the airport section like "fee per hour"

b.) If there isn´t a specified fee on that airport, use a "generic standard fee". That will be applied to all airports. We may could classify airports into different classes in order to set these fees. For example: 

Class A (small aerodromes)
Class B (middle airports)
Class C (big airports)
.....and so on...

Could something like that be possible to do?

Cheers!

**Comment by nabeelio** [comment]
@avs-code still interested?

**Comment by avs-code** [comment]
Yes, if possible, especially the minutes setup (calculate the time (in minutes) at the gate/parking from the acars).

The disposable add-on added a lot of economy-related features, but the time setting in minutes is still necessary. Also, if this is something that can be included in the core, that would be better, or at least configurable by administrators, also in minutes.

That's how they charge for services such as the use of the jetway and parking time. (#776 )
